tfhe/src/integer/server_key/radix_parallel/tests_cases_unsigned.rs

```diff
@@ -0,0 +1,2825 @@
+}
+
+pub(crate) fn unchecked_scalar_left_shift_test<P, T>(param: P, mut executor: T)
+where
+    P: Into<PBSParameters>,
+    T: for<'a> FunctionExecutor<(&'a RadixCiphertext, u64), RadixCiphertext>,
+{
+    let (cks, sks) = KEY_CACHE.get_from_params(param);
+    let cks = RadixClientKey::from((cks, NB_CTXT));
+
+    let mut rng = rand::thread_rng();
+
+    // message_modulus^vec_length
+    let modulus = cks.parameters().message_modulus().0.pow(NB_CTXT as u32) as u64;
+    let nb_bits = modulus.ilog2();
+
+    executor.setup(&cks, sks);
+
+    for _ in 0..NB_TEST {
+        let clear = rng.gen::<u64>() % modulus;
+        let scalar = rng.gen::<u32>();
+
+        let ct = cks.encrypt(clear);
+
+        // case when 0<= scalar < nb_bits
+        {
+            let scalar = scalar % nb_bits;
+            let encrypted_result = executor.execute((&ct, scalar as u64));
+            let decrypted_result: u64 = cks.decrypt(&encrypted_result);
+            assert_eq!(clear.wrapping_shl(scalar) % modulus, decrypted_result);
+        }
+
+        // case when scalar >= nb_bits
+        {
+            let scalar = scalar.saturating_add(nb_bits);
+            let encrypted_result = executor.execute((&ct, scalar as u64));
+            let decrypted_result: u64 = cks.decrypt(&encrypted_result);
+            assert_eq!(
+                clear.wrapping_shl(scalar % nb_bits) % modulus,
+                decrypted_result
+            );
+        }
+    }
+
+    let clear = rng.gen::<u64>() % modulus;
+    let ct = cks.encrypt(clear);
+
+    let nb_bits_in_block = cks.parameters().message_modulus().0.ilog2();
+    for scalar in 0..nb_bits_in_block {
+        let encrypted_result = executor.execute((&ct, scalar as u64));
+        let decrypted_result: u64 = cks.decrypt(&encrypted_result);
+        assert_eq!(clear.wrapping_shl(scalar) % modulus, decrypted_result);
+    }
+}
+
+pub(crate) fn unchecked_scalar_right_shift_test<P, T>(param: P, mut executor: T)
+where
+    P: Into<PBSParameters>,
+    T: for<'a> FunctionExecutor<(&'a RadixCiphertext, u64), RadixCiphertext>,
+{
+    let (cks, sks) = KEY_CACHE.get_from_params(param);
+    let cks = RadixClientKey::from((cks, NB_CTXT));
+
+    let mut rng = rand::thread_rng();
+
+    // message_modulus^vec_length
+    let modulus = cks.parameters().message_modulus().0.pow(NB_CTXT as u32) as u64;
+    let nb_bits = modulus.ilog2();
+
+    executor.setup(&cks, sks);
+
+    for _ in 0..NB_TEST {
+        let clear = rng.gen::<u64>() % modulus;
+        let scalar = rng.gen::<u32>();
+
+        let ct = cks.encrypt(clear);
+
+        // case when 0<= scalar < nb_bits
+        {
+            let scalar = scalar % nb_bits;
+            let encrypted_result = executor.execute((&ct, scalar as u64));
+            assert!(encrypted_result.block_carries_are_empty());
+            let decrypted_result: u64 = cks.decrypt(&encrypted_result);
+            assert_eq!(
+                clear.checked_shr(scalar).unwrap_or(0) % modulus,
```

Comment:
I think its some left overs when our shift was made so that when `shift >= nb_bits` it would return 0, which is what ` clear.checked_shr(scalar).unwrap_or(0)` does.

Will update